<!DOCTYPE html>
<html class="no-js" lang="en-us">
<head>
	<meta charset="UTF-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Landscape Genetic Simulations - NALgen</title>
	<script>(function(d,e){d[e]=d[e].replace("no-js","js");})(document.documentElement,"className");</script>
	<meta name="description" content="">
	<meta property="og:title" content="Landscape Genetic Simulations" />
<meta property="og:description" content="To evaluate the NALgen method of modeling neutral and adaptive genetic variation across continuous landscapes, I first generated several landscapes, and then simulated genotypes based on the generated landscapes. In this post, I break down the simulation steps." />
<meta property="og:type" content="article" />
<meta property="og:url" content="/NALgen/post/landscape-genetic-simulations/" />
<meta property="article:published_time" content="2019-07-17T00:00:00&#43;00:00"/>
<meta property="article:modified_time" content="2019-07-17T00:00:00&#43;00:00"/>

	
	<link rel="dns-prefetch" href="//fonts.googleapis.com">
	<link rel="dns-prefetch" href="//fonts.gstatic.com">
	<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700">
	<link rel="stylesheet" href="/NALgen/css/style.css">
	
	<link rel="shortcut icon" href="/NALgen/favicon.ico">
		
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-139514839-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</head>
<body class="body">
	<div class="container container--outer">
		<header class="header">
	<div class="container">
		<div class="logo">
			<a class="logo__link" href="/NALgen" title="NALgen" rel="home">
				<div class="logo__title">NALgen</div>
				<div class="logo__tagline">Neutral and Adaptive Landscape Genetics</div>
			</a>
		</div>
		
<nav class="menu">
	<button class="menu__btn" aria-haspopup="true" aria-expanded="false" tabindex="0">
		<span class="menu__btn-title" tabindex="-1">Menu</span>
	</button>
	<ul class="menu__list">
		<li class="menu__item">
			<a class="menu__link" href="/NALgen/about/">About</a>
		</li>
		<li class="menu__item">
			<a class="menu__link" href="/NALgen/vignettes/">Vignettes</a>
		</li>
		<li class="menu__item">
			<a class="menu__link" href="/NALgen/tutorials/">Tutorials</a>
		</li>
	</ul>
</nav>

	</div>
</header>
		<div class="wrapper flex">
			<div class="primary">
			
<main class="main" role="main">
	<article class="post">
		<header class="post__header">
			<h1 class="post__title">Landscape Genetic Simulations</h1>
			<div class="post__meta meta">
<div class="meta__item-datetime meta__item">
	<svg class="meta__icon icon icon-time" width="16" height="14" viewBox="0 0 30 28"><path d="M15 0C7 0 1 6 1 14s6 14 14 14 14-6 14-14S23 0 15 0zm0 25C9 25 4 20 4 14S9 3 15 3s11 5 11 11-5 11-11 11zm1-18h-2v8.4l6.8 4.4L22 18l-6-3.8V7z"/></svg>
	<time class="meta__text" datetime="2019-07-17T00:00:00">July 17, 2019</time>
</div>

<div class="meta__item-categories meta__item">
	<svg class="meta__icon icon icon-category" width="16" height="16" viewBox="0 0 16 16"><path d="m7 2l1 2h8v11h-16v-13z"/></svg>
	<span class="meta__text"><a class="meta__link" href="/NALgen/categories/simulations" rel="category">Simulations</a></span>
</div>
</div>
		</header><div class="content post__content clearfix">
			<p>To evaluate the NALgen method of modeling neutral and adaptive genetic variation across continuous landscapes, I first generated several landscapes, and then simulated genotypes based on the generated landscapes. In this post, I break down the simulation steps.</p>

<p><a href="https://github.com/chazhyseni/NALgen/tree/master/data" target="_blank">Simulated genotypes (50 selected and 500 neutral biallelic loci (SNPs), separately and combined): two migration coefficients (m=0.1 and 0.5) x two levels of landscape complexity (low and high) x two levels of landscape permeability (low and high)</a></p>

<h2 id="color-palette">Color palette</h2>

<pre><code class="language-r">library(ggsci)
#schwifty &lt;- pal_rickandmorty()
#schwifty.cols &lt;- colorRampPalette(schwifty(12)[c(3,12,4,6,1,9,8,2)])
#cols &lt;- schwifty.cols(10)
futr &lt;- pal_futurama()
futr.cols &lt;- colorRampPalette(futr(12)[c(12,11,3,7,8,6,1,2)])
cols &lt;- futr.cols(12)

library(scales)
#show_col(schwifty(12))
#show_col(cols)
#show_col(futr(12))
show_col(cols)
</code></pre>

<p><strong>Color palette:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-1-1.png" alt="" /><!-- --></p>

<h1 id="phase-i-test-single-locus-genotypes">Phase I (Test): Single-locus genotypes</h1>

<h2 id="generating-simple-low-complexity-landscapes">Generating simple (low-complexity) landscapes</h2>

<p><em>landscapeR</em> package: simulating simple landscapes (barrier and two different values of carrying capacity/habitat suitability): landscape 1 has low permeability cells (carrying capacity = 5) between habitable patches (carrying capacity = 100), while landscape 2 has high permeability cells (carrying capacity = 25), in addition to barriers (carrying capacity = 0).</p>

<pre><code class="language-r">library(landscapeR)
library(raster)
#consistency of results
set.seed(49137)
m &lt;- matrix(0, 26, 34)
r &lt;- raster(m, xmn=-91, xmx=-74, ymn=28, ymx=41)
r &lt;- makeClass(r, 20, 20, val=1)
r &lt;- makeClass(r, 20, 20, val=2)
r &lt;- rmSingle(r)
lclp &lt;- r
lclp[lclp==1] &lt;- 100
lclp[lclp==2] &lt;- 5
plot(lclp,breaks=seq(0,100,length.out=11),col=cols)
</code></pre>

<p><strong>Low-complexity low-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-2-1.png" alt="" /><!-- --></p>

<pre><code class="language-r">lchp &lt;- r
lchp[lchp==2] &lt;- 100
lchp[lchp==1] &lt;- 25
plot(lchp,breaks=seq(0,100,length.out=11),col=cols)
</code></pre>

<p><strong>Low-complexity high-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-2-2.png" alt="" /><!-- --></p>

<h2 id="simulation-of-genotypes-based-on-simple-landscapes">Simulation of genotypes based on simple landscapes</h2>

<p><em>landsim</em> package: simulating <strong>selected</strong> SNPs based on the <em>landscapeR</em>-generated landscape.</p>

<ul>
<li>Landscape 1: Low Complexity, <strong>Low</strong> Permeability</li>
</ul>

<pre><code class="language-r">library(landsim)
#consistency of results
set.seed(3356737)
pop1 &lt;- make_population(
                 habitat = lclp,
                 inaccessible.value = &quot;NA&quot;,
                 uninhabitable.value = 0,
                 genotypes = c(&quot;aa&quot;, &quot;aA&quot;, &quot;AA&quot;),
                 N = c(10, 10, 10)
             )

migr &lt;- migration(
                 kern = &quot;gaussian&quot;,
                 sigma = 10,
                 radius = 50,
                 normalize = 1
             )

migr1 &lt;- setup_migration(migr, pop1)

sel &lt;- c(1.01, 1.02, 1.03)

for (i in 1:3){
  germ.vital1 &lt;- vital(
               function (N, ...) {
                  out &lt;- r0 / (1 + migrate(rowSums(N),competition)/carrying.capacity)
                  return(cbind(aa=out, aA=s*out, AA=s^2*out))
               },
               r0 = 0.1,
               s = sel[i],  # multiplicative selective benefit of the A allele
               carrying.capacity = values(pop1$habitat)[pop1$habitable],
               competition = migr
           )

  germ.vital1 &lt;- setup_vital(germ.vital1, pop1)

  demog1 &lt;- demography(
          prob.seed = 0.2,
          fecundity = 80,
          prob.germination = germ.vital1,
          prob.survival = 0.8, 
          seed.migration = migr,
          pollen.migration = migr,
            genotypes = c(&quot;aa&quot;, &quot;aA&quot;, &quot;AA&quot;)
       )

  demog1 &lt;- setup_demography(demog1, pop1)

  plot.times &lt;- seq(0, 1000, length.out=6)

  sim1 &lt;- simulate_pop(pop1, demog1, times = plot.times, 
                carrying.capacity = values(pop1$habitat)[pop1$habitable],
                summaries = list(totals=function(N){colSums(N)}))

  matplot(sim1$summaries[[&quot;totals&quot;]], type='l', xlab='time', ylab='numbers', lty=1, 
          main=paste(&quot;selection coefficient =&quot;, sel[i]-1))
  legend(&quot;topright&quot;, lty=1, col=1:3, legend=colnames(sim1$summaries[[&quot;totals&quot;]]))
}
</code></pre>

<ul>
<li>Landscape 2: Low Complexity, <strong>High</strong> Permeability</li>
</ul>

<pre><code class="language-r">library(landsim)
#consistency of results
set.seed(635137)
pop2 &lt;- make_population(
                 habitat = lchp,
                 inaccessible.value = &quot;NA&quot;,
                 uninhabitable.value = 0,
                 genotypes = c(&quot;aa&quot;, &quot;aA&quot;, &quot;AA&quot;),
                 N = c(10, 10, 10)
             )

migr &lt;- migration(
                 kern = &quot;gaussian&quot;,
                 sigma = 10,
                 radius = 50,
                 normalize = 1
             )

migr2 &lt;- setup_migration(migr, pop2)

sel &lt;- c(1.01, 1.02, 1.03)

for (i in 1:3){
  germ.vital2 &lt;- vital(
               function (N, ...) {
                  out &lt;- r0 / (1 + migrate(rowSums(N),competition)/carrying.capacity)
                  return(cbind(aa=out, aA=s*out, AA=s^2*out))
               },
               r0 = 0.1,
               s = sel[i],  # multiplicative selective benefit of the A allele
               carrying.capacity = values(pop2$habitat)[pop2$habitable],
               competition = migr
           )

  germ.vital2 &lt;- setup_vital(germ.vital2, pop2)

  demog2 &lt;- demography(
          prob.seed = 0.2,
          fecundity = 80,
          prob.germination = germ.vital2,
          prob.survival = 0.8, 
          seed.migration = migr,
          pollen.migration = migr,
            genotypes = c(&quot;aa&quot;, &quot;aA&quot;, &quot;AA&quot;)
       )

  demog2 &lt;- setup_demography(demog2, pop2)

  plot.times &lt;- seq(0, 1000, length.out=6)

  sim2 &lt;- simulate_pop(pop2, demog2, times = plot.times, 
                carrying.capacity = values(pop2$habitat)[pop2$habitable],
                summaries = list(totals=function(N){colSums(N)}))

  matplot(sim2$summaries[[&quot;totals&quot;]], type='l', xlab='time', ylab='numbers', lty=1, 
          main=paste(&quot;selection coefficient =&quot;, sel[i]-1))
  legend(&quot;topright&quot;, lty=1, col=1:3, legend=colnames(sim2$summaries[[&quot;totals&quot;]]))
}
</code></pre>

<h1 id="phase-ii-multi-locus-genotypes">Phase II: Multi-locus genotypes</h1>

<h2 id="generating-simple-low-complexity-landscapes-1">Generating simple (low-complexity) landscapes</h2>

<p><em>landscapeR</em> package: simulating simple landscapes (barrier and two different values of carrying capacity/habitat suitability): landscape 1 has low permeability cells (carrying capacity = 5) between habitable patches (carrying capacity = 100), while landscape 2 has high permeability cells (carrying capacity = 25), in addition to barriers (carrying capacity = 0).</p>

<pre><code class="language-r">library(landscapeR)
library(raster)
#same seed = same landscapes as above
set.seed(49137)
m &lt;- matrix(0, 26, 34)
r &lt;- raster(m, xmn=-91, xmx=-74, ymn=28, ymx=41)
r &lt;- makeClass(r, 20, 20, val=1)
r &lt;- makeClass(r, 20, 20, val=2)
r &lt;- rmSingle(r)
lclp &lt;- r
lclp[lclp==1] &lt;- 100
lclp[lclp==2] &lt;- 5
plot(lclp,breaks=seq(0,100,length.out=11),col=cols)
</code></pre>

<p><strong>Low-complexity low-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-5-1.png" alt="" /><!-- --></p>

<pre><code class="language-r">lchp &lt;- r
lchp[lchp==2] &lt;- 100
lchp[lchp==1] &lt;- 25
plot(lchp,breaks=seq(0,100,length.out=11),col=cols)
</code></pre>

<p><strong>Low-complexity high-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-5-2.png" alt="" /><!-- --></p>

<h2 id="simulation-of-genotypes-based-on-simple-landscapes-1">Simulation of genotypes based on simple landscapes</h2>

<p><em>popRange</em> package: simulating 50 <strong>selected</strong> and 500 <strong>neutral</strong> SNPs based on the <em>landscapeR</em>-generated landscape.</p>

<pre><code class="language-r">library(popRange)
#consistency of results
set.seed(6531841)

barrier &lt;- lchp
barrier[barrier&lt;1] &lt;- -1
barrier[barrier&gt;1] &lt;- 1
#plot(lclp,breaks=seq(minValue(lclp),maxValue(lclp),length.out=11),col=cols)
#plot(lchp,breaks=seq(minValue(lchp),maxValue(lchp),length.out=11),col=cols)
#plot(barrier,breaks=seq(minValue(barrier),maxValue(barrier),length.out=11),col=cols)

lclp.Sel &lt;- lclp
lclp.Sel[lclp.Sel&lt;11] &lt;- 0
lclp.Sel[lclp.Sel&gt;10] &lt;- 0.01
lclp.Sel.df &lt;- cbind(rep(1,nrow(lclp.Sel)*ncol(lclp.Sel)),
                                  rep(50,nrow(lclp.Sel)*ncol(lclp.Sel)),
                                  as.data.frame(lclp.Sel),
                                  coordinates(lclp.Sel)-0.5)
colnames(lclp.Sel.df) &lt;- c(&quot;sSNP&quot;,&quot;fSNP&quot;,&quot;s&quot;,&quot;x&quot;,&quot;y&quot;)

lchp.Sel &lt;- lchp
lchp.Sel[lchp.Sel&lt;11] &lt;- 0
lchp.Sel[lchp.Sel&gt;10] &lt;- 0.01
lchp.Sel.df &lt;- cbind(rep(1,nrow(lchp.Sel)*ncol(lchp.Sel)),
                                  rep(50,nrow(lchp.Sel)*ncol(lchp.Sel)),
                                  as.data.frame(lchp.Sel),
                                  coordinates(lchp.Sel)-0.5)
colnames(lchp.Sel.df) &lt;- c(&quot;sSNP&quot;,&quot;fSNP&quot;,&quot;s&quot;,&quot;x&quot;,&quot;y&quot;)

s.coords=c()
for(i in 1:26) s.coords &lt;- c(s.coords, paste0(seq(0,25)[i], seq(0,33)))
s.coeffs &lt;- c()
for (i in 1:26) s.coeffs &lt;- c(s.coeffs, as.matrix(lclp.Sel)[i,])
sDiff.lclp &lt;- rbind(c(1,50,s.coeffs),
                    c(51,550,rep(0,884)))

s.coords=c()
for(i in 1:26) s.coords &lt;- c(s.coords, paste0(seq(0,25)[i], seq(0,33)))
s.coeffs &lt;- c()
for (i in 1:26) s.coeffs &lt;- c(s.coeffs, as.matrix(lchp.Sel)[i,])
sDiff.lchp &lt;- rbind(c(1,50,s.coeffs),
                    c(51,550,rep(0,884)))

#folders where simulations will go
m0.1_lclp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Landscapes/500neutral+50selectedSNPs/Low_Complexity_Landscapes/landscapeR/Low_Permeability&quot;
m0.5_lclp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Landscapes/500neutral+50selectedSNPs/Low_Complexity_Landscapes/landscapeR/Low_Permeability&quot;
m0.1_lchp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Landscapes/500neutral+50selectedSNPs/Low_Complexity_Landscapes/landscapeR/High_Permeability&quot;
m0.5_lchp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Landscapes/500neutral+50selectedSNPs/Low_Complexity_Landscapes/landscapeR/High_Permeability&quot;

setwd(m0.1_lclp)
popRangeSim(world=as.matrix(barrier), popSize=20, K=as.matrix(lclp), 
              diploid=TRUE, nGens=1000, mig=0.1, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.lclp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;lclp&quot;, GENEPOP=TRUE)
writeRaster(lclp, &quot;lclp.asc&quot;, overwrite=TRUE)

setwd(m0.1_lchp)
popRangeSim(world=as.matrix(barrier), popSize=20, K=as.matrix(lchp), 
              diploid=TRUE, nGens=1000, mig=0.1, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.lchp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;lchp&quot;, GENEPOP=TRUE)
writeRaster(lchp, &quot;lchp.asc&quot;, overwrite=TRUE)

setwd(m0.5_lclp)
popRangeSim(world=as.matrix(barrier), popSize=20, K=as.matrix(lclp), 
              diploid=TRUE, nGens=1000, mig=0.5, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.lclp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;lclp&quot;, GENEPOP=TRUE)
writeRaster(lclp, &quot;lclp.asc&quot;, overwrite=TRUE)

setwd(m0.5_lchp)
popRangeSim(world=as.matrix(barrier), popSize=20, K=as.matrix(lchp), 
              diploid=TRUE, nGens=1000, mig=0.5, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.lchp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;lchp&quot;, GENEPOP=TRUE)
writeRaster(lchp, &quot;lchp.asc&quot;, overwrite=TRUE)
</code></pre>

<h2 id="simulation-of-genotypes-on-complex-landscapes-generated-based-on-bioclim-data">Simulation of genotypes on complex landscapes generated based on Bioclim data</h2>

<p><em>virtualspecies</em> package: landscape generation based on Bioclim data:<br />
   To generate distributions:<br />
     1. collinearity removed from climatic variables,<br />
     2. PCA of remaining variables performed,<br />
     3. ranges of PCA values specified (for three metapopulations).</p>

<pre><code class="language-r">library(raster)
library(virtualspecies)
#consistency of results
set.seed(946516341)

setwd(&quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Phylogeography/Geo_Analysis/Data/EnvData/bioclim/east coast/current&quot;)

fn &lt;- list.files(pattern=&quot;.asc&quot;)
stk &lt;- stack()
for(i in 1:length(fn)) stk &lt;- addLayer(stk,raster(fn[i]))

nampres &lt;- sub(pattern=&quot;Ecoast&quot;,replacement=&quot;&quot;,names(stk))
nampres &lt;- sub(pattern=&quot;_&quot;,replacement=&quot;&quot;,nampres)
nampres &lt;- sub(pattern=&quot;_&quot;,replacement=&quot;&quot;,nampres)

names(stk) &lt;- nampres

stk60 &lt;- aggregate(stk,60)
rc_stk60 &lt;- removeCollinearity(stk60, select.variables = TRUE, plot = TRUE, multicollinearity.cutoff = 0.5)
</code></pre>

<p><strong>Selecting non-collinear bioclimatic variables</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-1.png" alt="" /><!-- --></p>

<pre><code class="language-r">#plot parameters
par(mfrow=c(1,1),fg=&quot;gray50&quot;,pty='m',bty='o',mar=c(4,4,4,4),cex.main=1.3,cex.axis=1.1,cex.lab=1.2)


##Low Permeability##

PCA_distrib_N &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(3.5,2.5), sds=c(1,1), plot=FALSE)
PCA_distrib_C &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(0,-2.2), sds=c(1,1), plot=FALSE)
PCA_distrib_S &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(-2.5,2.5), sds=c(1,1), plot=FALSE)


spN &lt;- convertToPA(PCA_distrib_N, species.prevalence=NULL, alpha=-0.002, beta=0.2)
</code></pre>

<p><strong>Occurrence probability of metapopulation 1 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-2.png" alt="" /><!-- --></p>

<pre><code class="language-r">spC &lt;- convertToPA(PCA_distrib_C, species.prevalence=NULL, alpha=-0.002, beta=0.2)
</code></pre>

<p><strong>Occurrence probability of metapopulation 2 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-3.png" alt="" /><!-- --></p>

<pre><code class="language-r">spS &lt;- convertToPA(PCA_distrib_S, species.prevalence=NULL, alpha=-0.002, beta=0.2)
</code></pre>

<p><strong>Occurrence probability of metapopulation 3 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-4.png" alt="" /><!-- --></p>

<pre><code class="language-r">pop3.complex.lowperm.PA &lt;- 1*spN$pa.raster + 2*spC$pa.raster + 3*spS$pa.raster
pop3.complex.lowperm.PA &lt;- round(pop3.complex.lowperm.PA, 0)
pop3.complex.lowperm.PA[pop3.complex.lowperm.PA&gt;3] &lt;- 0
pop3.complex.lowperm.PA[is.na(pop3.complex.lowperm.PA)] &lt;- 0

spN$suitab.raster[spN$suitab.raster&lt;0.25] &lt;- 0
spC$suitab.raster[spC$suitab.raster&lt;0.25] &lt;- 0
spS$suitab.raster[spS$suitab.raster&lt;0.25] &lt;- 0

spN_suitab &lt;- 100*spN$pa.raster*spN$suitab.raster
spC_suitab &lt;- 100*spC$pa.raster*spC$suitab.raster
spS_suitab &lt;- 100*spS$pa.raster*spS$suitab.raster

pop3.complex.lowperm.Suitab &lt;- spN_suitab + spC_suitab + spS_suitab
pop3.complex.lowperm.Suitab[pop3.complex.lowperm.Suitab&gt;100] &lt;- 100
pop3.complex.lowperm.Suitab[is.na(pop3.complex.lowperm.Suitab)] &lt;- 0
pop3.complex.lowperm.Suitab[pop3.complex.lowperm.Suitab&lt;5] &lt;- 0
pop3.complex.lowperm.Suitab &lt;- round(pop3.complex.lowperm.Suitab,0)

plot(pop3.complex.lowperm.Suitab,
    breaks=seq(minValue(pop3.complex.lowperm.Suitab),maxValue(pop3.complex.lowperm.Suitab),length.out=11),
    col=cols)
</code></pre>

<p><strong>Putting the three metapopulations together: High-complexity low-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-5.png" alt="" /><!-- --></p>

<pre><code class="language-r">pop3.complex.lowperm.Barrier &lt;- pop3.complex.lowperm.Suitab
pop3.complex.lowperm.Barrier[pop3.complex.lowperm.Suitab&gt;24] &lt;- 1
pop3.complex.lowperm.Barrier[pop3.complex.lowperm.Suitab&lt;25] &lt;- -1

pop3.complex.lowperm.catProb &lt;- pop3.complex.lowperm.Barrier
pop3.complex.lowperm.catProb[pop3.complex.lowperm.catProb==-1] &lt;- 1
pop3.complex.lowperm.catProb[pop3.complex.lowperm.catProb==1] &lt;- 0.0001

pop3.complex.lowperm.Sel &lt;- pop3.complex.lowperm.PA
pop3.complex.lowperm.Sel[pop3.complex.lowperm.Sel==1] &lt;- 0.01
pop3.complex.lowperm.Sel[pop3.complex.lowperm.Sel==2] &lt;- 0.02
pop3.complex.lowperm.Sel[pop3.complex.lowperm.Sel==3] &lt;- 0.03
pop3.complex.lowperm.Sel.df &lt;- cbind(rep(1,nrow(pop3.complex.lowperm.Sel)*ncol(pop3.complex.lowperm.Sel)),
                                  rep(50,nrow(pop3.complex.lowperm.Sel)*ncol(pop3.complex.lowperm.Sel)),
                                  as.data.frame(pop3.complex.lowperm.Sel),
                                  coordinates(pop3.complex.lowperm.Sel))
colnames(pop3.complex.lowperm.Sel.df)=c(&quot;start&quot;,&quot;end&quot;,&quot;s&quot;,&quot;x&quot;,&quot;y&quot;)


##High Permeability##

PCA_distrib_N &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(2.5,1), sds=c(1.2,1.2), plot=FALSE)
PCA_distrib_C &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(-0.5,-0.5), sds=c(1.2,1.2), plot=FALSE)
PCA_distrib_S &lt;- generateSpFromPCA(raster.stack=stk60[[rc_stk60]], axes=c(1:2), sample.points=TRUE, nb.points=850, means=c(-2.5,2), sds=c(1.2,1.2), plot=FALSE)

spN &lt;- convertToPA(PCA_distrib_N, species.prevalence=NULL, alpha=-0.004, beta=0.4)
</code></pre>

<p><strong>Occurrence probability of metapopulation 1 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-6.png" alt="" /><!-- --></p>

<pre><code class="language-r">spC &lt;- convertToPA(PCA_distrib_C, species.prevalence=NULL, alpha=-0.004, beta=0.4)
</code></pre>

<p><strong>Occurrence probability of metapopulation 2 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-7.png" alt="" /><!-- --></p>

<pre><code class="language-r">spS &lt;- convertToPA(PCA_distrib_S, species.prevalence=NULL, alpha=-0.004, beta=0.4)
</code></pre>

<p><strong>Occurrence probability of metapopulation 3 (out of 3):</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-8.png" alt="" /><!-- --></p>

<pre><code class="language-r">pop3.complex.highperm.PA &lt;- 1*spN$pa.raster + 2*spC$pa.raster + 3*spS$pa.raster
pop3.complex.highperm.PA &lt;- round(pop3.complex.highperm.PA, 0)
pop3.complex.highperm.PA[pop3.complex.highperm.PA&gt;3] &lt;- 0
pop3.complex.highperm.PA[is.na(pop3.complex.highperm.PA)] &lt;- 0

spN$suitab.raster[spN$suitab.raster&lt;0.25] &lt;- 0
spC$suitab.raster[spC$suitab.raster&lt;0.25] &lt;- 0
spS$suitab.raster[spS$suitab.raster&lt;0.25] &lt;- 0

spN_suitab &lt;- 100*spN$pa.raster*spN$suitab.raster
spC_suitab &lt;- 100*spC$pa.raster*spC$suitab.raster
spS_suitab &lt;- 100*spS$pa.raster*spS$suitab.raster

pop3.complex.highperm.Suitab &lt;- spN_suitab + spC_suitab + spS_suitab
pop3.complex.highperm.Suitab[pop3.complex.highperm.Suitab&gt;100] &lt;- 100
pop3.complex.highperm.Suitab[is.na(pop3.complex.highperm.Suitab)] &lt;- 0
pop3.complex.highperm.Suitab[pop3.complex.highperm.Suitab&lt;5] &lt;- 0
pop3.complex.highperm.Suitab &lt;- round(pop3.complex.highperm.Suitab,0)

plot(pop3.complex.highperm.Suitab,
    breaks=seq(minValue(pop3.complex.highperm.Suitab),maxValue(pop3.complex.highperm.Suitab),length.out=11),
    col=cols)
</code></pre>

<p><strong>Putting the three metapopulations together: High-complexity high-permeability landscape:</strong>
<img src="../../md-files/landgen-simulations/figure-html/unnamed-chunk-7-9.png" alt="" /><!-- --></p>

<pre><code class="language-r">pop3.complex.highperm.Barrier &lt;- pop3.complex.highperm.Suitab
pop3.complex.highperm.Barrier[pop3.complex.highperm.Suitab&gt;24] &lt;- 1
pop3.complex.highperm.Barrier[pop3.complex.highperm.Suitab&lt;25] &lt;- -1

pop3.complex.highperm.catProb &lt;- pop3.complex.highperm.Barrier
pop3.complex.highperm.catProb[pop3.complex.highperm.catProb==-1] &lt;- 1
pop3.complex.highperm.catProb[pop3.complex.highperm.catProb==1] &lt;- 0.0001  #see catProb value for popRange sims

pop3.complex.highperm.Sel &lt;- pop3.complex.highperm.PA
pop3.complex.highperm.Sel[pop3.complex.highperm.Sel==1] &lt;- 0.01
pop3.complex.highperm.Sel[pop3.complex.highperm.Sel==2] &lt;- 0.02
pop3.complex.highperm.Sel[pop3.complex.highperm.Sel==3] &lt;- 0.03
pop3.complex.highperm.Sel.df &lt;- cbind(rep(1,nrow(pop3.complex.highperm.Sel)*ncol(pop3.complex.highperm.Sel)),
                                  rep(50,nrow(pop3.complex.highperm.Sel)*ncol(pop3.complex.highperm.Sel)),
                                  as.data.frame(pop3.complex.highperm.Sel),
                                  coordinates(pop3.complex.highperm.Sel))
colnames(pop3.complex.highperm.Sel.df)=c(&quot;start&quot;,&quot;end&quot;,&quot;s&quot;,&quot;x&quot;,&quot;y&quot;)
</code></pre>

<p><em>popRange</em> package: simulating 50 <em>selected</em> and 500 <strong>neutral</strong> SNPs based on the <em>virtualspecies</em>-generated landscape.</p>

<pre><code class="language-r">library(popRange)
#consistency of results
set.seed(36813911)

hclp &lt;- pop3.complex.lowperm.Suitab
barrier.hclp &lt;- pop3.complex.lowperm.Barrier 
hchp &lt;- pop3.complex.highperm.Suitab
barrier.hchp &lt;- pop3.complex.highperm.Barrier 

#folders where simulations will go
m0.1_hclp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Landscapes/500neutral+50selectedSNPs/High_Complexity_Landscapes/virtualspecies/Low_Permeability&quot;
m0.5_hclp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Landscapes/500neutral+50selectedSNPs/High_Complexity_Landscapes/virtualspecies/Low_Permeability&quot;
m0.1_hchp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Landscapes/500neutral+50selectedSNPs/High_Complexity_Landscapes/virtualspecies/High_Permeability&quot;
m0.5_hchp &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Landscapes/500neutral+50selectedSNPs/High_Complexity_Landscapes/virtualspecies/High_Permeability&quot;

hclp.Sel &lt;- pop3.complex.lowperm.Sel
s.coords=c()
for(i in 1:26) s.coords &lt;- c(s.coords, paste0(seq(0,25)[i], seq(0,33)))
s.coeffs &lt;- c()
for (i in 1:26) s.coeffs &lt;- c(s.coeffs, as.matrix(hclp.Sel)[i,])
sDiff.hclp &lt;- rbind(c(1,50,s.coeffs),
                    c(51,550,rep(0,884)))

hchp.Sel &lt;- pop3.complex.highperm.Sel
s.coords=c()
for(i in 1:26) s.coords &lt;- c(s.coords, paste0(seq(0,25)[i], seq(0,33)))
s.coeffs &lt;- c()
for (i in 1:26) s.coeffs &lt;- c(s.coeffs, as.matrix(hchp.Sel)[i,])
sDiff.hchp &lt;- rbind(c(1,50,s.coeffs),
                    c(51,550,rep(0,884)))

setwd(m0.1_hclp)
popRangeSim(world=as.matrix(barrier.hclp), popSize=20, K=as.matrix(hclp), 
              diploid=TRUE, nGens=1000, mig=0.1, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.hclp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;hclp&quot;, GENEPOP=TRUE)
writeRaster(hclp, &quot;hclp.asc&quot;, overwrite=TRUE)

setwd(m0.1_hchp)
popRangeSim(world=as.matrix(barrier.hchp), popSize=20, K=as.matrix(hchp), 
              diploid=TRUE, nGens=1000, mig=0.1, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.hchp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;hchp&quot;, GENEPOP=TRUE)
writeRaster(hchp, &quot;hchp.asc&quot;, overwrite=TRUE)

setwd(m0.5_hclp)
popRangeSim(world=as.matrix(barrier.hclp), popSize=20, K=as.matrix(hclp), 
              diploid=TRUE, nGens=1000, mig=0.5, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.hclp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;hclp&quot;, GENEPOP=TRUE)
writeRaster(hclp, &quot;hclp.asc&quot;, overwrite=TRUE)

setwd(m0.5_hchp)
popRangeSim(world=as.matrix(barrier.hchp), popSize=20, K=as.matrix(hchp), 
              diploid=TRUE, nGens=1000, mig=0.5, 
              #gamma_shape=0.2, gamma_scale=1,
              #SNP_model=1, gSize=1*10^9, mutRate=1*10^-8,
              SNP_model=0, nSNPs=550, SNPs_starting_freq=0.5,
              #rMean=0.5, rVar=0.5,
              s=matrix(c(1,51,50,550,0.01,0),nrow=2,ncol=3),
              #sDiff=sDiff.hchp,
              #had to modify config.py float(catProb) instead of int(catProb)
              catProb=0.0001,
              outfile=&quot;hchp&quot;, GENEPOP=TRUE)
writeRaster(hchp, &quot;hchp.asc&quot;, overwrite=TRUE)
</code></pre>

<h1 id="data-for-further-analysis">Data for further analysis</h1>

<h2 id="subsets-of-simulated-genotypes">Subsets of simulated genotypes</h2>

<pre><code class="language-r">library(adegenet)
library(poppr)
library(zvau)

m0.1_n &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Genotypes/20pop_subsets/neutral_only&quot;
m0.1_n.s &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Genotypes/20pop_subsets/neutral+selected&quot;
m0.1_s &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_1/Simulated_Genotypes/20pop_subsets/selected_only&quot;
m0.5_n &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Genotypes/20pop_subsets/neutral_only&quot;
m0.5_n.s &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Genotypes/20pop_subsets/neutral+selected&quot;
m0.5_s &lt;- &quot;C:/Users/chazh/Documents/Research Projects/Reticulitermes/Simulations/popRange/m_0_5/Simulated_Genotypes/20pop_subsets/selected_only&quot;

setwd(m0.1_lclp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;lclp.GENEPOP.gen1000&quot;)[-2],&quot;lclp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;lclp.GENEPOP.PopCoor.gen1000&quot;),&quot;lclp.GENEPOP.PopCoor.txt&quot;)
lclp_file &lt;- read.genepop(paste0(lclp,&quot;lclp.GENEPOP.gen&quot;))
lclp_file@other &lt;- read.table(paste0(lclp,&quot;lclp.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(lclp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.lclp &lt;- popsub(lclp_file, sublist=sample(popNames(lclp_file),20))
rand.subset.20pops.lclp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.lclp)[,52:551],pop=genind2df(rand.subset.20pops.lclp)[,1],ncode=2)
rand.subset.20pops.lclp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.lclp)[,2:51],pop=genind2df(rand.subset.20pops.lclp)[,1],ncode=2)
setwd(m0.1_n.s)
writeGenPop(rand.subset.20pops.lclp,&quot;lclp_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_n)
writeGenPop(rand.subset.20pops.lclp.neutrloc,&quot;lclp_only-neutr_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_s)
writeGenPop(rand.subset.20pops.lclp.selloc,&quot;lclp_only-sel_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.1_lchp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;lchp.GENEPOP.gen1000&quot;)[-2],&quot;lchp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;lchp.GENEPOP.PopCoor.gen1000&quot;),&quot;lchp.GENEPOP.PopCoor.txt&quot;)
lchp_file &lt;- read.genepop(paste0(lchp,&quot;lchp.GENEPOP.gen&quot;))
lchp_file@other &lt;- read.table(paste0(lchp,&quot;lchp.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(lchp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.lchp &lt;- popsub(lchp_file, sublist=sample(popNames(lchp_file),20))
rand.subset.20pops.lchp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.lchp)[,52:551],pop=genind2df(rand.subset.20pops.lchp)[,1],ncode=2)
rand.subset.20pops.lchp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.lchp)[,2:51],pop=genind2df(rand.subset.20pops.lchp)[,1],ncode=2)
setwd(m0.1_n.s)
writeGenPop(rand.subset.20pops.lchp,&quot;lchp_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_n)
writeGenPop(rand.subset.20pops.lchp.neutrloc,&quot;lchp_only-neutr_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_s)
writeGenPop(rand.subset.20pops.lchp.selloc,&quot;lchp_only-sel_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.1_hclp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;hclp.GENEPOP.gen1000&quot;)[-2],&quot;hclp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;hclp.GENEPOP.PopCoor.gen1000&quot;),&quot;hclp.GENEPOP.PopCoor.txt&quot;)
hclp_file &lt;- read.genepop(paste0(hclp,&quot;pop3_complex_lowperm.GENEPOP.gen&quot;))
hclp_file@other &lt;- read.table(paste0(hclp,&quot;pop3_complex_lowperm.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(hclp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.hclp &lt;- popsub(hclp_file, sublist=sample(popNames(hclp_file),20))
rand.subset.20pops.hclp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.hclp)[,52:551],pop=genind2df(rand.subset.20pops.hclp)[,1],ncode=2)
rand.subset.20pops.hclp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.hclp)[,2:51],pop=genind2df(rand.subset.20pops.hclp)[,1],ncode=2)
setwd(m0.1_n.s)
writeGenPop(rand.subset.20pops.hclp,&quot;hclp_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_n)
writeGenPop(rand.subset.20pops.hclp.neutrloc,&quot;hclp_only-neutr_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_s)
writeGenPop(rand.subset.20pops.hclp.selloc,&quot;hclp_only-sel_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.1_hchp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;hchp.GENEPOP.gen1000&quot;)[-2],&quot;hchp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;hchp.GENEPOP.PopCoor.gen1000&quot;),&quot;hchp.GENEPOP.PopCoor.txt&quot;)
hchp_file &lt;- read.genepop(paste0(hchp,&quot;pop3_complex_highperm.GENEPOP.gen&quot;))
hchp_file@other &lt;- read.table(paste0(hchp,&quot;pop3_complex_highperm.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(hchp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.hchp &lt;- popsub(hchp_file, sublist=sample(popNames(hchp_file),20))
rand.subset.20pops.hchp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.hchp)[,52:551],pop=genind2df(rand.subset.20pops.hchp)[,1],ncode=2)
rand.subset.20pops.hchp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.hchp)[,2:51],pop=genind2df(rand.subset.20pops.hchp)[,1],ncode=2)
setwd(m0.1_n.s)
writeGenPop(rand.subset.20pops.hchp,&quot;hchp_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_n)
writeGenPop(rand.subset.20pops.hchp.neutrloc,&quot;hchp_only-neutr_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.1_s)
writeGenPop(rand.subset.20pops.hchp.selloc,&quot;hchp_only-sel_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)



setwd(m0.5_lclp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;lclp.GENEPOP.gen1000&quot;)[-2],&quot;lclp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;lclp.GENEPOP.PopCoor.gen1000&quot;),&quot;lclp.GENEPOP.PopCoor.txt&quot;)
lclp_file &lt;- read.genepop(paste0(lclp,&quot;lclp.GENEPOP.gen&quot;))
lclp_file@other &lt;- read.table(paste0(lclp,&quot;lclp.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(lclp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.lclp &lt;- popsub(lclp_file, sublist=sample(popNames(lclp_file),20))
rand.subset.20pops.lclp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.lclp)[,52:551],pop=genind2df(rand.subset.20pops.lclp)[,1],ncode=2)
rand.subset.20pops.lclp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.lclp)[,2:51],pop=genind2df(rand.subset.20pops.lclp)[,1],ncode=2)
setwd(m0.5_n.s)
writeGenPop(rand.subset.20pops.lclp,&quot;lclp_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_n)
writeGenPop(rand.subset.20pops.lclp.neutrloc,&quot;lclp_only-neutr_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_s)
writeGenPop(rand.subset.20pops.lclp.selloc,&quot;lclp_only-sel_20pops.gen&quot;,
            comment=&quot;Low Complexity Low Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lclp@other$x,rand.subset.20pops.lclp@other$y),&quot;lclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.5_lchp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;lchp.GENEPOP.gen1000&quot;)[-2],&quot;lchp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;lchp.GENEPOP.PopCoor.gen1000&quot;),&quot;lchp.GENEPOP.PopCoor.txt&quot;)
lchp_file &lt;- read.genepop(paste0(lchp,&quot;lchp.GENEPOP.gen&quot;))
lchp_file@other &lt;- read.table(paste0(lchp,&quot;lchp.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(lchp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.lchp &lt;- popsub(lchp_file, sublist=sample(popNames(lchp_file),20))
rand.subset.20pops.lchp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.lchp)[,52:551],pop=genind2df(rand.subset.20pops.lchp)[,1],ncode=2)
rand.subset.20pops.lchp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.lchp)[,2:51],pop=genind2df(rand.subset.20pops.lchp)[,1],ncode=2)
setwd(m0.5_n.s)
writeGenPop(rand.subset.20pops.lchp,&quot;lchp_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_n)
writeGenPop(rand.subset.20pops.lchp.neutrloc,&quot;lchp_only-neutr_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_s)
writeGenPop(rand.subset.20pops.lchp.selloc,&quot;lchp_only-sel_20pops.gen&quot;,
            comment=&quot;Low Complexity High Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.lchp@other$x,rand.subset.20pops.lchp@other$y),&quot;lchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.5_hclp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;hclp.GENEPOP.gen1000&quot;)[-2],&quot;hclp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;hclp.GENEPOP.PopCoor.gen1000&quot;),&quot;hclp.GENEPOP.PopCoor.txt&quot;)
hclp_file &lt;- read.genepop(paste0(hclp,&quot;pop3_complex_lowperm.GENEPOP.gen&quot;))
hclp_file@other &lt;- read.table(paste0(hclp,&quot;pop3_complex_lowperm.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(hclp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.hclp &lt;- popsub(hclp_file, sublist=sample(popNames(hclp_file),20))
rand.subset.20pops.hclp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.hclp)[,52:551],pop=genind2df(rand.subset.20pops.hclp)[,1],ncode=2)
rand.subset.20pops.hclp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.hclp)[,2:51],pop=genind2df(rand.subset.20pops.hclp)[,1],ncode=2)
setwd(m0.5_n.s)
writeGenPop(rand.subset.20pops.hclp,&quot;hclp_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_n)
writeGenPop(rand.subset.20pops.hclp.neutrloc,&quot;hclp_only-neutr_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_s)
writeGenPop(rand.subset.20pops.hclp.selloc,&quot;hclp_only-sel_20pops.gen&quot;,
            comment=&quot;High Complexity Low Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hclp@other$x,rand.subset.20pops.hclp@other$y),&quot;hclp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)


setwd(m0.5_hchp)
#delete duplicate &quot;Loc1&quot; line
writeLines(readLines(&quot;hchp.GENEPOP.gen1000&quot;)[-2],&quot;hchp.GENEPOP.gen&quot;)
writeLines(readLines(&quot;hchp.GENEPOP.PopCoor.gen1000&quot;),&quot;hchp.GENEPOP.PopCoor.txt&quot;)
hchp_file &lt;- read.genepop(paste0(hchp,&quot;pop3_complex_highperm.GENEPOP.gen&quot;))
hchp_file@other &lt;- read.table(paste0(hchp,&quot;pop3_complex_highperm.GENEPOP.PopCoor.txt&quot;), sep=&quot; &quot;,header=F)
colnames(hchp_file@other)=c(&quot;x&quot;,&quot;y&quot;)
rand.subset.20pops.hchp &lt;- popsub(hchp_file, sublist=sample(popNames(hchp_file),20))
rand.subset.20pops.hchp.neutrloc &lt;- df2genind(genind2df(rand.subset.20pops.hchp)[,52:551],pop=genind2df(rand.subset.20pops.hchp)[,1],ncode=2)
rand.subset.20pops.hchp.selloc &lt;- df2genind(genind2df(rand.subset.20pops.hchp)[,2:51],pop=genind2df(rand.subset.20pops.hchp)[,1],ncode=2)
setwd(m0.5_n.s)
writeGenPop(rand.subset.20pops.hchp,&quot;hchp_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_n)
writeGenPop(rand.subset.20pops.hchp.neutrloc,&quot;hchp_only-neutr_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - Only Neutral Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
setwd(m0.5_s)
writeGenPop(rand.subset.20pops.hchp.selloc,&quot;hchp_only-sel_20pops.gen&quot;,
            comment=&quot;High Complexity High Permeability - Only Selected Loci - 20 pops - Simulated Genotypes&quot;)
write.table(cbind(rand.subset.20pops.hchp@other$x,rand.subset.20pops.hchp@other$y),&quot;hchp_20pops_coords.txt&quot;, row.names=FALSE, col.names=FALSE)
</code></pre>

<p><a href="https://github.com/chazhyseni/NALgen/tree/master/data" target="_blank">Simulated genotypes (50 selected and 500 neutral biallelic loci (SNPs), separately and combined): two migration coefficients (m=0.1 and 0.5) x two levels of landscape complexity (low and high) x two levels of landscape permeability (low and high)</a></p>
		</div>
		
<div class="post__tags tags clearfix">
	<svg class="tags__icon icon icon-tag" width="16" height="16" viewBox="0 0 32 32"><path d="M32 19c0 1-1 2-1 2L21 31s-1 1-2 1-2-1-2-1L2 16c-1-1-1.4-2-1.4-2S0 12.5 0 11V3C0 1.5.8.8.8.8S1.5 0 3 0h8c1.5 0 3 .6 3 .6S15 1 16 2l15 15s1 1 1 2zM7 10a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"/></svg>
	<ul class="tags__list">
		<li class="tags__item"><a class="tags__link btn" href="/NALgen/tags/landscape-genetics/" rel="tag">Landscape Genetics</a></li>
		<li class="tags__item"><a class="tags__link btn" href="/NALgen/tags/landscape-simulations/" rel="tag">Landscape Simulations</a></li>
		<li class="tags__item"><a class="tags__link btn" href="/NALgen/tags/genotype-simulations/" rel="tag">Genotype Simulations</a></li>
	</ul>
</div>
	</article>
</main>

<div class="authorbox clearfix">
	<figure class="authorbox__avatar">
		<img alt="Chaz Hyseni avatar" src="/NALgen/img/SpEcEvGen.png" class="avatar" height="90" width="90">
	</figure>
	<div class="authorbox__header">
		<span class="authorbox__name">About Chaz Hyseni</span>
	</div>
	<div class="authorbox__description">
		Chaz has been doing research in the field of Population Genetics since 2006. He is interested in the distribution of genetic variation (both neutral and adaptive) in natural environments, at different spatial and temporal scales (from Landscape Genetics to Phylogeography). To understand pattern and process, Chaz uses empirical data as well as spatial eco-evolutionary modeling.
	</div>
</div>

<nav class="post-nav flex">
	<div class="post-nav__item post-nav__item--next">
		<a class="post-nav__link" href="/NALgen/post/multivariate-analysis_simulated-data/" rel="next"><span class="post-nav__caption">Next&thinsp;</span><p class="post-nav__post-title">NALgen Analysis Pt.1</p></a>
	</div>
</nav>


			</div>
			<aside class="sidebar">
<div class="widget-recent widget">
	<h4 class="widget__title">Recent Posts</h4>
	<div class="widget__content">
		<ul class="widget__list">
			<li class="widget__item"><a class="widget__link" href="/NALgen/post/multivariate-analysis_simulated-data/">NALgen Analysis Pt.1</a></li>
			<li class="widget__item"><a class="widget__link" href="/NALgen/post/landscape-genetic-simulations/">Landscape Genetic Simulations</a></li>
		</ul>
	</div>
</div>
<div class="widget-categories widget">
	<h4 class="widget__title">Categories</h4>
	<div class="widget__content">
		<ul class="widget__list">
			<li class="widget__item"><a class="widget__link" href="/NALgen/categories/analysis">Analysis</a></li>
			<li class="widget__item"><a class="widget__link" href="/NALgen/categories/simulations">Simulations</a></li>
		</ul>
	</div>
</div>
<div class="widget-taglist widget">
	<h4 class="widget__title">Tags</h4>
	<div class="widget__content">
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/co-inertia-analysis" title="Co inertia analysis">Co inertia analysis</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/discriminant-analysis-of-principal-components" title="Discriminant analysis of principal components">Discriminant analysis of principal components</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/genetic-environmental-covariance" title="Genetic environmental covariance">Genetic environmental covariance</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/genotype-simulations" title="Genotype simulations">Genotype simulations</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/landscape-genetics" title="Landscape genetics">Landscape genetics</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/landscape-simulations" title="Landscape simulations">Landscape simulations</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/multivariate-ordination" title="Multivariate ordination">Multivariate ordination</a>
		<a class="widget-taglist__link widget__link btn" href="/NALgen/tags/population-structure" title="Population structure">Population structure</a>
	</div>
</div>
</aside>
		</div>
		<footer class="footer">
	<div class="container footer__container flex">
		
		<div class="footer__copyright">
			&copy; 2019 NALgen.
			<span class="footer__copyright-credits">Generated with <a href="https://gohugo.io/" rel="nofollow noopener" target="_blank">Hugo</a> and <a href="https://github.com/Vimux/Mainroad/" rel="nofollow noopener" target="_blank">Mainroad</a> theme.</span>
		</div>
	</div>
</footer>
	</div>
<script async defer src="/NALgen/js/menu.js"></script></body>
</html>