<!DOCTYPE html>
<!-- Created by pdf2htmlEX (https://github.com/coolwanglu/pdf2htmlex) -->
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8"/>

<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
<link rel="stylesheet" href="base.min.css"/>
<link rel="stylesheet" href="fancy.min.css"/>
<link rel="stylesheet" href="main.css"/>
<script src="compatibility.min.js"></script>
<script src="theViewer.min.js"></script>
<script>
try{
theViewer.defaultViewer = new theViewer.Viewer({});
}catch(e){}
</script>
<title></title>
</head>
<body>
<div id="sidebar">
<div id="outline">
</div>
</div>
<div id="page-container">
<div id="pf1" class="pf w0 h0" data-page-no="1"><div class="pc pc1 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg1.png"/><div class="t m0 x1 h1 y1 ff1 fs0 fc0 sc0 ls0 ws0">NALgen</div><div class="t m0 x2 h2 y2 ff2 fs1 fc0 sc0 ls1 ws0">NALgen (Neutral and Adaptive Landscape Genetics) is a method for modeling </div><div class="t m0 x3 h2 y3 ff2 fs1 fc0 sc0 ls1 ws0">the process of gene flow and predicting the resulting pattern of (neutral and </div><div class="t m0 x4 h2 y4 ff2 fs1 fc0 sc0 ls1 ws0">adaptive) genetic variation across (continuous) landscapes.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2" class="pf w0 h0" data-page-no="2"><div class="pc pc2 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg2.png"/><div class="t m0 x5 h3 y5 ff1 fs2 fc0 sc0 ls0 ws0">NALgen:</div><div class="t m0 x5 h3 y6 ff1 fs2 fc0 sc0 ls0 ws0">Analysis Steps</div><div class="t m0 x6 h4 y7 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff4 fs4 fc2">To evaluate the NALgen method of modeling neutral and adaptive genetic variation across </span></div><div class="t m0 x7 h5 y8 ff4 fs4 fc2 sc0 ls1 ws0">continuous landscapes:</div><div class="t m0 x8 h4 y9 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc2">Generated several landscapes, and then simulated genotypes based on the generated landscapes</span></div><div class="t m0 x8 h4 ya ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc2">Used multivariate analyses </span></div><div class="t m0 x9 h4 yb ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">to infer population structure in the simulated data </span></div><div class="t m0 x9 h4 yc ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">to find covariance between genetic and environmental data. </span></div><div class="t m0 x8 h4 yd ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc2">Modeled neutral and adaptive genetic variation across continuous landscapes</span></div><div class="t m0 x9 h4 ye ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">Information about population structure used to create response variable</span></div><div class="t m0 x9 h4 yf ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">Transformed environmental data (based on genetic-environmental covariance) used as </span></div><div class="t m0 xa h6 y10 ff2 fs4 fc2 sc0 ls1 ws0">predictors</div><div class="t m0 x8 h4 y11 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc2">Evaluated performance of model (using subsets of 20 randomly-distributed populations)</span></div><div class="t m0 x9 h4 y12 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">Compared against original landscape used to simulate genotypes</span></div><div class="t m0 x9 h4 y13 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc2">Compared against inferred map of ancestry based on genotypes simulated for each cell </span></div><div class="t m0 xa h6 y14 ff2 fs4 fc2 sc0 ls1 ws0">(all 884 populations)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3" class="pf w0 h0" data-page-no="3"><div class="pc pc3 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg3.png"/><div class="t m0 xb h3 y15 ff1 fs2 fc2 sc0 ls0 ws0">Landscape </div><div class="t m0 xc h3 y16 ff1 fs2 fc2 sc0 ls0 ws0">Genetic </div><div class="t m0 xd h3 y17 ff1 fs2 fc2 sc0 ls0 ws0">Simulations</div><div class="t m0 xe h4 y18 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff4 fs4 fc3">Generating simple (low-complexity) landscapes and simulating genotypes</span></div><div class="t m0 xf h4 y19 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc3">landscapeR<span class="ff2">package: simulating simple landscapes (barrier and two </span></span></div><div class="t m0 x10 h6 y1a ff2 fs4 fc3 sc0 ls1 ws0">different values of carrying capacity/habitat suitability): landscape 1 has </div><div class="t m0 x10 h6 y1b ff2 fs4 fc3 sc0 ls1 ws0">low permeability cells (carrying capacity = 5) between habitable patches </div><div class="t m0 x10 h6 y1c ff2 fs4 fc3 sc0 ls1 ws0">(carrying capacity = 100), while landscape 2 has high permeability cells </div><div class="t m0 x10 h6 y1d ff2 fs4 fc3 sc0 ls1 ws0">(carrying capacity = 25), in addition to barriers (carrying capacity = 0)</div><div class="t m0 xf h4 y1e ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc3">popRange<span class="ff2">package: simulating 50<span class="ff4">selected<span class="_ _1"></span></span>and 500<span class="ff4">neutral</span>SNPs based </span></span></div><div class="t m0 x10 h6 y1f ff2 fs4 fc3 sc0 ls1 ws0">on the<span class="ff5">landscapeR</span>-generated landscape.</div><div class="t m0 xe h4 y20 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff4 fs4 fc3">Generating high-complexity landscapes and simulating genotypes</span></div><div class="t m0 xf h4 y21 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc3">virtualspecies<span class="ff2">package: landscape generation based on Bioclim data</span></span></div><div class="t m0 x11 h4 y22 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc3">Collinearity removed from climatic variables,</span></div><div class="t m0 x11 h4 y23 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc3">PCA of remaining variables performed,</span></div><div class="t m0 x11 h4 y24 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff2 fs4 fc3">Ranges of PCA values specified (for three metapopulations).</span></div><div class="t m0 xf h4 y25 ff3 fs3 fc1 sc0 ls1 ws0"><span class="_ _0"> </span><span class="ff5 fs4 fc3">popRange<span class="ff2">package: simulating 50</span>selected<span class="ff2">and 500<span class="ff4">neutral</span>SNPs based on </span></span></div><div class="t m0 x10 h6 y26 ff2 fs4 fc3 sc0 ls1 ws0">the<span class="ff5">virtualspecies</span>-generated landscape.</div><div class="t m0 x12 h7 y27 ff4 fs5 fc4 sc0 ls1 ws0">Simulated genotypes (50 selected and 500 neutral biallelic loci (SNPs), separately and combined): two migration coefficients </div><div class="t m0 x12 h7 y28 ff4 fs5 fc4 sc0 ls1 ws0">(m=0.1 and 0.5) x two levels of landscape complexity (low and high) x two levels of landscape permeability (low and high)</div><a class="l" href="https://github.com/chazhyseni/NALgen/tree/master/data"><div class="d m1" style="border-style:none;position:absolute;left:49.152740px;bottom:42.995953px;width:856.009735px;height:16.440430px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="https://github.com/chazhyseni/NALgen/tree/master/data"><div class="d m1" style="border-style:none;position:absolute;left:49.152740px;bottom:26.195953px;width:831.981415px;height:16.440430px;background-color:rgba(255,255,255,0.000001);"></div></a></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf4" class="pf w0 h0" data-page-no="4"><div class="pc pc4 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg4.png"/><div class="t m0 x13 h3 y29 ff1 fs2 fc5 sc0 ls0 ws0">Low-Complexity </div><div class="t m0 x13 h3 y2a ff1 fs2 fc5 sc0 ls0 ws0">Low-Permeability </div><div class="t m0 x13 h3 y2b ff1 fs2 fc5 sc0 ls0 ws0">(LCLP) landscape</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf5" class="pf w0 h0" data-page-no="5"><div class="pc pc5 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg5.png"/><div class="t m0 x13 h3 y29 ff1 fs2 fc5 sc0 ls0 ws0">Low-Complexity </div><div class="t m0 x13 h3 y2a ff1 fs2 fc5 sc0 ls0 ws0">High-Permeability </div><div class="t m0 x13 h3 y2b ff1 fs2 fc5 sc0 ls0 ws0">(LCHP) landscape</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf6" class="pf w0 h0" data-page-no="6"><div class="pc pc6 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg6.png"/><div class="t m0 x13 h3 y29 ff1 fs2 fc5 sc0 ls0 ws0">High-Complexity </div><div class="t m0 x13 h3 y2a ff1 fs2 fc5 sc0 ls0 ws0">Low-Permeability </div><div class="t m0 x13 h3 y2b ff1 fs2 fc5 sc0 ls0 ws0">(HCLP) landscape</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf7" class="pf w0 h0" data-page-no="7"><div class="pc pc7 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg7.png"/><div class="t m0 x13 h3 y29 ff1 fs2 fc5 sc0 ls0 ws0">High-Complexity </div><div class="t m0 x13 h3 y2a ff1 fs2 fc5 sc0 ls0 ws0">High-Permeability </div><div class="t m0 x13 h3 y2b ff1 fs2 fc5 sc0 ls0 ws0">(HCHP) landscape</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf8" class="pf w0 h0" data-page-no="8"><div class="pc pc8 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg8.png"/><div class="t m0 x14 h3 y2c ff1 fs2 fc3 sc0 ls0 ws0">Setup of Evaluation Step: </div><div class="t m2 x15 h8 y2d ff6 fs6 fc3 sc0 ls2 ws0">884-population reference</div><div class="t m0 x16 h9 y2e ff2 fs7 fc3 sc0 ls1 ws0">The original landscapes used to </div><div class="t m0 x17 h9 y2f ff2 fs7 fc3 sc0 ls1 ws0">simulate the genetic data, as well </div><div class="t m0 x18 h9 y30 ff2 fs7 fc3 sc0 ls1 ws0">as population structure results </div><div class="t m0 x18 h9 y31 ff2 fs7 fc3 sc0 ls1 ws0">obtained from these complete </div><div class="t m0 x19 h9 y32 ff2 fs7 fc3 sc0 ls1 ws0">genetic datasets, will be used in </div><div class="t m0 x16 h9 y33 ff2 fs7 fc3 sc0 ls1 ws0">evaluating the NALgen method.</div><div class="t m0 x1a h9 y2e ff2 fs7 fc3 sc0 ls1 ws0">Presented hereinafter are the </div><div class="t m0 x1b h9 y2f ff2 fs7 fc3 sc0 ls1 ws0">reference results (mapped </div><div class="t m0 x1c h9 y30 ff2 fs7 fc3 sc0 ls1 ws0">ancestry coefficients) that NALgen </div><div class="t m0 x1d h9 y31 ff2 fs7 fc3 sc0 ls1 ws0">modeling based on 20 randomly-</div><div class="t m0 x1e h9 y32 ff2 fs7 fc3 sc0 ls1 ws0">distributed populations will be </div><div class="t m0 x1f h9 y33 ff2 fs7 fc3 sc0 ls1 ws0">compared against. </div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf9" class="pf w0 h0" data-page-no="9"><div class="pc pc9 w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bg9.png"/></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfa" class="pf w0 h0" data-page-no="a"><div class="pc pca w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bga.png"/></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfb" class="pf w0 h0" data-page-no="b"><div class="pc pcb w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgb.png"/></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfc" class="pf w0 h0" data-page-no="c"><div class="pc pcc w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgc.png"/></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pfd" class="pf w0 h0" data-page-no="d"><div class="pc pcd w0 h0"><img class="bi x0 y0 w0 h0" alt="" src="bgd.png"/><div class="t m0 x20 h3 y34 ff1 fs2 fc5 sc0 ls0 ws0">NALgen Method</div><div class="t m0 x21 h9 y35 ff2 fs7 fc3 sc0 ls1 ws0">To determine the maximum genetic-</div><div class="t m0 x21 h9 y36 ff2 fs7 fc3 sc0 ls1 ws0">environmental covariance, the raw data is </div><div class="t m0 x21 h9 y37 ff2 fs7 fc3 sc0 ls1 ws0">first transformed: these transformed </div><div class="t m0 x21 h9 y38 ff2 fs7 fc3 sc0 ls1 ws0">environmental scores form the predictors </div><div class="t m0 x21 h9 y39 ff2 fs7 fc3 sc0 ls1 ws0">in NALgen modeling</div><div class="t m0 x21 h9 y3a ff2 fs7 fc3 sc0 ls1 ws0">The response variables are measures of </div><div class="t m0 x21 h9 y3b ff2 fs7 fc3 sc0 ls1 ws0">genetic variation (neutral and adaptive): </div><div class="t m0 x21 h9 y3c ff2 fs7 fc3 sc0 ls1 ws0">these are newly-developed metrics</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
</div>
<div class="loading-indicator">

</div>
</body>
</html>
